src/main/kotlin/app/revanced/util/bytecode/BytecodeUtils.kt

```diff
@@ -0,0 +1,43 @@
+package app.revanced.util.bytecode
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.util.proxy.mutableTypes.MutableClass
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.instruction.BuilderInstruction21c
+import org.jf.dexlib2.iface.ClassDef
+import org.jf.dexlib2.iface.Method
+
+
+internal object BytecodeUtils{
+    public fun transformIns(context: BytecodeContext, transformFn: (Ins: Any, index: Int, methodDef: Method, classDef: ClassDef, context: BytecodeContext) -> Any, opcodes: List<Opcode>) {
+        context.transformInstructions(transformFn, opcodes, context)
+    }
+
+
+    private fun BytecodeContext.transformInstructions(transformFn: (Ins: Any, index: Int, methodDef: Method, classDef: ClassDef, context: BytecodeContext) -> Any , opcodes: List<Opcode>, context:BytecodeContext) {
```

Comment:
Iirc such a method exists in the utils package.

Reply:
where?

Reply:
In `app.revanced.util.patch`.

Reply:
if you are talking about [this](https://github.com/andrisas/revanced-patches/blob/20b2612efdcf08e6dacfced87bc0e6b580e85146/src/main/kotlin/app/revanced/util/patch/AbstractTransformInstructionsPatch.kt) it can not filter for opcodes.  Should i move/rename the one that i made?

Reply:
It is possible to filter for opcodes. Here is an example: https://github.com/revanced/revanced-patches/blob/43d22014c75e0845b506f50eb1e14e996e7d9272/src/main/kotlin/app/revanced/util/patch/MethodCall.kt#L83-L85

It is used by the `filterMap` function as you can see here:
https://github.com/revanced/revanced-patches/blob/43d22014c75e0845b506f50eb1e14e996e7d9272/src/main/kotlin/app/revanced/patches/all/connectivity/wifi/spoof/patch/SpoofWifiPatch.kt#L192-L202